packages/apps/graph/src/graph/query/fungible-accounts-by-public-key.ts

```diff
@@ -0,0 +1,109 @@
+import { prismaClient } from '@db/prisma-client';
+import { getFungibleChainAccount } from '@services/account-service';
+import { chainIds } from '@utils/chains';
+import { normalizeError } from '@utils/errors';
+import { builder } from '../builder';
+import FungibleAccount from '../objects/fungible-account';
+import type { FungibleChainAccount } from '../types/graphql-types';
+import { FungibleAccountName } from '../types/graphql-types';
+
+builder.queryField('fungibleAccountsByPublicKey', (t) =>
+  t.field({
+    description: 'Retrieve an account by public key.',
+    args: {
+      publicKey: t.arg.string({
+        required: true,
+        validate: {
+          minLength: 1,
+        },
+      }),
+      fungibleName: t.arg.string({
+        required: false,
+        validate: {
+          minLength: 1,
+        },
+      }),
+    },
+    type: [FungibleAccount],
+    nullable: true,
+    async resolve(__parent, args) {
+      try {
+        const accountAndFungibleNames =
+          await getAccountAndFungibleNamesByPublicKey(
+            args.publicKey,
+            args.fungibleName ?? undefined,
+          );
+
+        if (accountAndFungibleNames.length === 0) {
+          return null;
```

Comment:
In terms of consistency, for all the other queries that return arrays, do we return an empty array on no results or null?

Reply:
Good catch. It is indeed an array so we should return an empty array and make the query non-nullable. Changes done 👍🏽